src/import.php

```diff
@@ -37,6 +37,346 @@
+    foreach ($zScheduledFiles as $sFile => $zScheduledFile) {
+        if (!isset($aFiles[0][$sFile]) && !isset($aFiles[1][$sFile])) {
+            $aFiles[1][$sFile] =
+                '0,' .
+                (9 - $zScheduledFile['priority']) . ',' .
+                (!$zScheduledFile['processed_date']? '0000-00-00 00:00:00' : $zScheduledFile['processed_date']) . ',' .
+                 $zScheduledFile['scheduled_date'] . ',' .
+                 $sFile;
+        }
+    }
+
+    if (!$zScheduledFiles && !$aFiles[0] && !$aFiles[1]) {
+        // No files found, and nothing scheduled!
+        lovd_showInfoTable('No files found, and nothing scheduled!<BR>Please check the documentation to see how to import LOVD files directly from the server.', 'information');
+        $_T->printFooter();
+        exit;
+    }
+
+
+
+
+
+    // Process POST if present.
+    $nScheduled = 0;
+    if (POST && !empty($_POST['files_to_schedule'])) {
+        // We can't use the file names as keys, as PHP replaces dots with underscores and then we can't find the files reliably anymore.
+        // Loop through variables, find them in $aFiles[0], mark them in the database as scheduled.
+        foreach ($_POST['files_to_schedule'] as $sFile) {
+            if (isset($aFiles[0][$sFile])) {
+                // Process this file, it's present indeed and not yet imported.
+                if ($_DB->query('INSERT IGNORE INTO ' . TABLE_SCHEDULED_IMPORTS . ' (filename, scheduled_by, scheduled_date) VALUES (?, ?, NOW())', array($sFile, $_AUTH['id']))->rowCount()) {
+                    // Scheduled successfully, didn't exist yet.
+                    $nScheduled ++;
+                    lovd_writeLog('Event', LOG_EVENT, 'Scheduled ' . $sFile . ' for import');
+
+                    // Now, to facilitate proper sorting, refresh the information in the files array.
+                    // FIXME: This assumes a priority of 0 (stored as 9); we need to change this when we implement priority scheduling.
+                    $aFiles[0][$sFile] = '0,9,0000-00-00 00:00:00,' . date('Y-m-d H:i:s') . substr($aFiles[0][$sFile], 43);
+
+                    // Also, to display statistics, load the file's info.
+                    // This includes the filename field, that we usually don't have, but whatever.
+                    $zScheduledFiles[$sFile] = $_DB->query('SELECT sf.*, u.name AS scheduled_by_name
+                                                            FROM ' . TABLE_SCHEDULED_IMPORTS . ' AS sf LEFT OUTER JOIN ' . TABLE_USERS . ' AS u ON (sf.scheduled_by = u.id)
+                                                            WHERE sf.filename = ?', array($sFile))->fetchAssoc();
+                }
+            }
+        }
+    }
+
+
+
+
+
+    // Sort the file list (the DB list is already sorted).
+    asort($aFiles[0]); // Sort data files, keeping indices.
+    asort($aFiles[1]); // Sort data files, keeping indices.
+    $nFilesProcessed = count($aFiles[1]);
```

Comment:
The code that builds the string to sort files with is IMO hard to understand. I think storing info about the files in $aFiles as separate values and using usort() here with a user-defined comparison function. E.g. this would make hacking the sort value on line 161 more readable (simply change a few values instead of using substr()).

Reply:
You're right, I should have thought about that. I'll make the changes, it'll take more code but it will make things much simpler to understand if I need to revise this later.

Reply:
Implemented in e5bf63b. I'll make a few other changes, and will ask you to have another look, if you don't mind.